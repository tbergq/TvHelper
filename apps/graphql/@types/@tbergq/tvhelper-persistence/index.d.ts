declare module '@tbergq/tvhelper-persistence';
