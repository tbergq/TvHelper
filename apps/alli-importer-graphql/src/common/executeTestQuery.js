// @flow

import { graphql } from 'graphql';

import schema from '../Schema';

const createContext = () => ({});

export default function executeTestQuery(
  query: string,
  variables: ?{ ... },
  request: ?{ ... } = {},
): Promise<{ ... }> {
  // $FlowExpectedError: Ok for testing purposes
  return graphql(schema, query, null, createContext(request), variables);
}
