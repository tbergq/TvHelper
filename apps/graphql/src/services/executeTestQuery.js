// @flow

import { graphql } from 'graphql';

import schema from '../Schema';
import createContext from './createGraphqlContext';

export default function executeTestQuery(
  query: string,
  variables: ?{ ... },
  request: ?{ ... } = {},
): Promise<{ ... }> {
  // $FlowExpectedError: Ok for testing purposes
  return graphql(schema, query, null, createContext(request), variables);
}
