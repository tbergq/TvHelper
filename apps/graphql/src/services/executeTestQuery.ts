import { graphql } from 'graphql';

import schema from '../Schema';
import createContext from './createGraphqlContext';

export default function executeTestQuery(
  query: string,
  variables: {} | null | undefined,
  request: {} | null | undefined = {},
): Promise<{}> {
  // @ts-ignore: Ok for testing purposes
  return graphql(schema, query, null, createContext(request), variables);
}
