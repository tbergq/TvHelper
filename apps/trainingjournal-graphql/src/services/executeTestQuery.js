// @flow

import { graphql } from 'graphql';

import schema from '../Schema';

export default function executeTestQuery(query: string, variables: ?{ ... }): Promise<{ ... }> {
  return graphql(schema, query, null, null, variables);
}
