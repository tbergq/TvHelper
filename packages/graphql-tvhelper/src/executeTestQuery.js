// @flow

import { graphql as originalGraphQL, GraphQLSchema, GraphQLObjectType } from 'graphql';

import SearchTvShow from './search/queries/SearchTvShow';

const RootQuery = new GraphQLObjectType({
  name: 'RootQuery',
  description: 'Root Query',
  fields: {
    searchTvShow: SearchTvShow,
  },
});

const schema = new GraphQLSchema({
  query: RootQuery,
});

/**
 * Executes GraphQL query using our schema. This function should
 * be used only during testing because it uses faked context.
 */
export default function executeTestQuery(
  query: string,
  variables: ?{ ... },
  context: ?{ ... },
): Promise<Object> {
  return originalGraphQL(schema, query, null, context, variables);
}
