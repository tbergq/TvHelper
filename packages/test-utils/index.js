// @flow

export { default as withMongoDb } from './src/useMongoDb';
