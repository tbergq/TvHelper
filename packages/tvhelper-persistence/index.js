// @flow

import Favorite from './src/dataObjects/Favorite';

// Repositories
export { default as UserRepository } from './src/repositories/User';
export { default as FavoritesRepository } from './src/repositories/Favorites';
export { default as WatchedEpisodeRepository } from './src/repositories/WatchedEpisode';

// Models
export { default as UserModel } from './src/models/UserModel';
export { default as FavoritesModel } from './src/models/Favorites';

// types
export type FavoriteType = Favorite;

// Connection
export { default as tvHelperConnection } from './src/connection';
