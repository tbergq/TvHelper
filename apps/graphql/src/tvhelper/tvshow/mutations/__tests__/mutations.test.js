// @flow

import { tvHelperConnection, UserRepository } from '@tbergq/tvhelper-persistence';
import { generateTestsFromFixtures } from '@kiwicom/test-utils';

import executeTestQuery from '../../../../services/executeTestQuery';

let userId;

describe('AddFavorite', () => {
  beforeEach(async () => {
    const user = await UserRepository.createUser({
      username: 'lol',
      password: 'lol',
      email: 'lol@lol.no',
    });
    userId = user.id;
  }, 600000);

  afterEach(async () => {
    await tvHelperConnection.collection('users').drop();
  });
  generateTestsFromFixtures(`${__dirname}/__fixtures__`, input =>
    executeTestQuery(input, null, { user: { id: userId } }),
  );
});
