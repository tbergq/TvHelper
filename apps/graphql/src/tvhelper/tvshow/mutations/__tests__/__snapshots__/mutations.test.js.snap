// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddFavorite matches expected output: AddFavorite.graphql 1`] = `
~~~~~~~~~~ INPUT ~~~~~~~~~~
mutation addFavorite {
  addFavorite(serieId: "dHZzaG93OjY=") {
    success
    tvShow {
      node {
        name
        status
      }
    }
  }
}
~~~~~~~~~~ OUTPUT ~~~~~~~~~~
{
  "data": {
    "addFavorite": {
      "success": true,
      "tvShow": {
        "node": {
          "name": "The 100",
          "status": "Running"
        }
      }
    }
  }
}
`;
